File: template.rs
use std::env;
use std::io::Read;

fn part1() {
    let input = read_input();
    println!("{:?}", input);
    panic!("part1 not implemented!")
}

fn part2() {
    let input = read_input();
    println!("{:?}", input);
    panic!("part2 not implemented!")
}

fn read_input() -> String {
    let mut input = String::new();
    std::io::stdin().read_to_string(&mut input).unwrap();
    input
}

fn main() {
    let args: Vec<String> = env::args().collect();
    match args.len() {
        2 => match &args[1][..] {
            "1" => part1(),
            "2" => part2(),
            _ => panic!("invalid argument {}\n", args[1]),
        },
        _ => panic!(
            "invalid number of arguments {}; must provide 2\n",
            args.len()
        ),
    }
}

File: template.py
#!/usr/bin/env python

import sys


def part_one(data: str) -> int:
    print(data)
    raise NotImplementedError("part_one")


def part_two(data: str) -> int:
    print(data)
    raise NotImplementedError("part_two")


def read_input() -> str:
    return sys.stdin.read().strip("\n")


def main(args) -> None:
    data = read_input()

    match a := args[0]:
        case "1":
            print(f"part_one: {part_one(data)}")
        case "2":
            print(f"part_two: {part_two(data)}")
        case _:
            raise ValueError(f"invalid argument '{a}'") 


if __name__ == "__main__":
    main(sys.argv[1:])

